Subject: [directfb] fix X11 detection
From: Erwin Rol <erwin@erwinrol.com>
X-Git-Url: http://git.directfb.org/?p=core%2FDirectFB.git;a=blobdiff_plain;f=configure.in;fp=configure.in;h=6b7e0bacb1615dc56afffc2d00b6999b11327613;hp=1ca4d6d7cb925eef48b259af5e372c79b90c615d;hb=2dad4f2dde329f4bfc0e9e658b7171ea16d080dd;hpb=b49b12c58b8df83b8127cdebf20c426a70f1bb1b

Use pkgconfig for X11 detection. This diff is simply cherrypicked from
upstream and should not be needed anymore for the next release
of directfb.

Signed-off-by: Erwin Rol <erwin@erwinrol.com>

---

diff --git a/configure.in b/configure.in
index 1ca4d6d..6b7e0ba 100644
--- a/configure.in
+++ b/configure.in
@@ -134,18 +134,8 @@ AC_ARG_ENABLE(x11,
   enable_x11=yes)
 
 if test "$enable_x11" = "yes"; then
-  CFLAGS_saved="$CFLAGS"
-  CFLAGS="$CFLAGS -I/usr/X11R6/include"
-  AC_CHECK_HEADER(X11/X.h, x11_found=yes, x11_found=no)
-  CFLAGS="$CFLAGS_saved"
-  if test "$x11_found" = no; then
-    enable_x11=no
-    AC_MSG_WARN([
-*** no X11/X.h found -- building without X11 support.])
-  else
-    X11_LIBS="-L/usr/X11R6/lib -lX11 -lXext"
-    X11_CFLAGS="-I/usr/X11R6/include"
-  fi
+  PKG_CHECK_MODULES([X11], [xproto x11 xext], [enable_x11="yes"], [enable_x11="no",
+    AC_MSG_WARN([*** no X11 found -- building without X11 support])])
 fi
 
 AM_CONDITIONAL(X11_CORE, test "$enable_x11" = "yes")
